src/tests/GfxBatchHbaoTest.cpp

```diff
@@ -39,70 +39,130 @@ namespace Cr = Corrade;
 namespace Mn = Magnum;
 using namespace Mn::Math::Literals;
 
+// Change this if we want to test on a different core mesh. Requires
+// regenerating ground truth images.
 constexpr const char* baseTestFilename = "van-gogh-room";
 
-const std::string SourceTestPlyDir =
+// Location of test scene mesh from habitat_test_secenes repository.
+const Cr::Containers::String testPlySourceDir =
     Cr::Utility::Path::join(SCENE_DATASETS, "habitat-test-scenes");
 
-const std::string TestHBAOImageDir =
+// Validation images location (held in repository
+// TODO : consider moving these images to habitat_test_secenes repository.
+const Cr::Containers::String testHBAOImageDir =
     Cr::Utility::Path::join(TEST_ASSETS, "hbao_tests");
 
+// Size of images to test for validation
 constexpr Mn::Vector2i Size{320, 240};
 
-constexpr const float aspectRatio = float(Size.x()) / Size.y();
+// Aspect ratio of given size
+constexpr const float AspectRatio = float(Size.x()) / Size.y();
 
-struct TestDataType;
 struct Projection;
+struct TestDataType;
 
 struct GfxBatchHbaoTest : Mn::GL::OpenGLTester {
   explicit GfxBatchHbaoTest();
   void generateTestData();
-  void testPerspective();
 
-  void testPerspectiveFlipped();
-  void testOrthographic();
+  Cr::Containers::Optional<Mn::GL::Texture2D> loadImageIntoTexture();
+
+  /// @brief Validation Tests ///
 
+  /**
+   * @brief Test standard orientation HBAO effect
+   * @param data The pertinent test data for the test
+   * @param filename The name of the validation file to test against
+   * @param proj The projection matrix this test consumes
+   */
   void testHBAOData(const TestDataType& data,
-                    const std::string& filename,
+                    const Cr::Containers::String& filename,
```

Comment:
Since you're making the changes already, function arguments should always be just string views (as opposed to return values, which usually need to be full owned strings). And it's a small view (just two pointers) so it can be passed as a value:

```suggestion
                    Cr::Containers::StringView filename,
```

Same below.